<a href="https://colab.research.google.com/github/ShabuddinDhafedar/Airbnb-Bookings-Analysis/blob/main/Airbnb_Bookings_Analysis_Capstone_Project(shabuddin).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **What is Exploratory Data Analysis and Why it is necessary?**

I’m referring to Terence S, a data scientist, on his explanation of exploratory data analysis.To simplify, exploratory data analysis(EDA), also known as Data Exploration is a step in the Data Analysis process, where several techniques are used to better understand the dataset being used.

**Some of the techniques are:-**

Extracting important variables and leaving behind useless variables Identifying missing values, and human error Understanding the data,outliers, maximizing our insight on a dataset and minimizing potential error that may occur later in the process By conducting EDA, we can turn an almost useable or unusable dataset into a useable dataset.

**Main components of Exploratory Data Analysis:-**

1.Collecting and loading data.

2.Processing and Cleaning dataset.

3.Analyzing and Visualizing Data.

# **Why Airbnb?**

Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions, understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more.

# **1.Collecting and loading data**

For this project, we are using Google colab a web IDE with a python programming language to write our script. IDE or Integrated Development Environment is a software application used for software development. To get the data, we are using Airbnb data that publicly shared on the internet under the Creative Commons License. Before we are able to load the data into our IDE, first we need to import various external libraries/modules that needed for visualization and analysis.

**a. Load python libraries**

*  Pandas and Numpy library used for data analysis

*  Matplotlib and Seaborn library used for data visualization


In [ ]:
#Importing libraries whichever needed 
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.image as mpimg
import seaborn as sns 
from datetime import datetime
%matplotlib inline


**b. Load dataset**

To load the dataset, we use pandas library and function to read the CSV file Airbnb Since 2008 dataset from dirve , convert it to the DataFrame and check the top 5 index data.

In [ ]:
#Mounting the drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
#Read of airbnb file
airbnb_df = pd.read_csv("/content/drive/MyDrive/Almabetter /Team Capstone Projects/ Exploratory Data Analysis (Choose any one)/Airbnb Bookings Analysis/Airbnb NYC 2019.csv")
airbnb_df.head()

id  ... availability_365
0  2539  ...              365
1  2595  ...              355
2  3647  ...              365
3  3831  ...              194
4  5022  ...                0

[5 rows x 16 columns]

**c. Understanding data**

After loading the dataset, Understanding the dataset by using various techniques.We have to look data type of every column if it already matches our requirement. For instance, we need a numerical data type,it may be integer or float, on the longitude,latitude,price,minimum nights,number of reviews,calculated host listings count and availability365,for listing names we need to make sure the data is using string/object data type.And also we need to look for information on how big is our dataset. By using shape attributes, we get to know our data size from a number of rows which consist of listing index, and the number of columns with the content of every features related to the index.

In [ ]:
#Checking the data
airbnb_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [ ]:
#Checking the data shape
airbnb_df.shape

(48895, 16)

Here we have found out that our dataset has 48895 listings and 16columns,which includes id,name,host id, location information, location coordinate, room type, the price, and so on.